File: JS/rental-app-js/services/rentalService.js
const rentals = [];

function rentItem(item, startDate, endDate) {
  for (const r of rentals) {
    if (r.item.name === item.name &&
        !(endDate < r.startDate || startDate > r.endDate)) {
      return {
        success: false,
        message: `Item already rented from ${r.startDate.toDateString()} to ${r.endDate.toDateString()}`
      };
    }
  }

  rentals.push({ item, startDate, endDate });
  return { success: true };
}

function getActiveRentals() {
  return rentals.filter(r => r.endDate >= new Date());
}

module.exports = { rentItem, getActiveRentals };

File: JAVA/rental-app-java/src/services/RentalService.java
package services;

import models.Item;
import models.Rental;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;
import java.util.stream.Collectors;

public class RentalService {
    private final List<Rental> rentals = new ArrayList<>();

    public String rentItem(Item item, LocalDate start, LocalDate end) {
        for (Rental rental : rentals) {
            if (rental.getItem().getName().equalsIgnoreCase(item.getName()) &&
                    !(end.isBefore(rental.getStartDate()) || start.isAfter(rental.getEndDate()))) {
                return "Item already rented from " + rental.getStartDate() + " to " + rental.getEndDate();
            }
        }

        rentals.add(new Rental(item, start, end));
        return "Item rented successfully!";
    }

    public List<Rental> getActiveRentals() {
        return rentals.stream()
                .filter(Rental::isActive)
                .collect(Collectors.toList());
    }
}
